porcupine/plugins/filemanager.py

```diff
@@ -162,20 +161,24 @@ def paste(new_path: Path) -> None:
 
 
 def paste_here(new_path: Path) -> None:
-    assert copy_path is not None
+    assert copy_path is not None and copy_path.is_file()
     assert new_path.is_dir()
 
     new_file_path = new_path / copy_path.name
 
     if new_file_path.exists():
-        new_file_path = ask_name_for_renaming(new_file_path, is_paste=True)  # rename-overwrite file conflict
-        if new_file_path is None:
+        show_overwriting_option = True
+        if new_file_path.parent == copy_path.parent:
+            show_overwriting_option = False
+        path = ask_file_name(new_file_path, is_paste=True, show_overwriting_option=show_overwriting_option)
```

Comment:
A simpler way to view this: whether to show overwriting option is same as whether we are copying into a different directory.
```suggestion
        path = ask_file_name(new_file_path, is_paste=True, show_overwriting_option=(new_file_path.parent != copy_path.parent))
```
The resulting line of code is now a bit long, but `black` will split this up on several lines.